### PR TITLE
[AUD-1187] Replace kebab-menu with share-user-button

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- replaces kebab-menu with share-user-button
- refactors share-user-button to be separate component

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1187-replace-kebab-with-share-button

File: src/containers/profile-page/components/mobile/ProfilePage.module.css [hunk 1]
```diff
@@ -42,16 +42,4 @@
...
-  width: 40px;
-  display: flex;
-  justify-content: center;
-  align-items: center;
-}
-
-.shareButton svg {
-  height: 24px;
-}
+}
```
Comment (sliptype) [inline src/containers/profile-page/components/mobile/ProfilePage.module.css:45]: I think prettier may not have run here

File: src/containers/profile-page/components/mobile/ProfilePage.tsx [hunk 2]
```diff
@@ -320,19 +317,7 @@ const ProfilePage = g(
...
-            onClick={() => onShare()}
-          />
-        ) : (
-          <IconButton
-            className={styles.overflowNav}
-            icon={<IconKebabHorizontal />}
-            onClick={onClickOverflow}
-          />
-        )
+        rightNav = <ShareUserButton userId={userId} />
```
Comment (rickyrombo) [inline src/containers/profile-page/components/mobile/ProfilePage.tsx:320]: Are we getting rid of the option to follow/unfollow from the overflow menu?

File: src/containers/profile-page/components/mobile/ProfilePage.tsx [hunk 3]
```diff
@@ -349,8 +334,7 @@ const ProfilePage = g(
       onCancel,
       onSave,
       hasMadeEdit,
-      onClickOverflow,
-      onShare
+      onClickOverflow
```
Comment (rickyrombo) [inline src/containers/profile-page/components/mobile/ProfilePage.tsx:337]: Is this still a dep?

**Review by sliptype — APPROVED** [review]
Looks lgtm to me!

File: src/containers/profile-page/components/mobile/ProfilePage.tsx (hunk 2 above)
Comment (rickyrombo) [reply src/containers/profile-page/components/mobile/ProfilePage.tsx:320]: If so, we can remove the onClickOverflow defined above

File: src/containers/profile-page/components/mobile/ProfilePage.module.css (hunk 1 above)
Comment (dylanjeffers) [reply src/containers/profile-page/components/mobile/ProfilePage.module.css:45]: good catch ^^

File: src/containers/profile-page/components/mobile/ProfilePage.tsx (hunk 2 above)
Comment (dylanjeffers) [reply src/containers/profile-page/components/mobile/ProfilePage.tsx:320]: @rickyrombo yeah we are, good call!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1187-replace-kebab-with-share-button

**Review by rickyrombo — APPROVED** [review]

**Pull request merged**